### DISCUSSION TITLE
Anki 2.1.54 Csv Support

### DISCUSSION THREAD

**Discussion by jeprecated** [axelboc/anki-ultimate-geography#542] [General]
https://github.com/ankitects/anki-changes/commit/3bda402ed46d68dd0c9b8def8dc8b2a97657bde8

> A new CSV **importer & exporter**, thanks to Rumo:
> 
> - Shows a preview based on the selected field separator.
> - Column mapping includes field text for reference.
> - Multiple decks and notetypes can now be imported/exported at once, as columns can be mapped to deck or notetype name.
> - **Supports exporting and importing a unique identifier for a note**, to use instead of a matching first field. This removes the need for the 'add note id' add-on.
> - Extends the header syntax so that files can specify metadata such as the desired field separator, **and the mapping of fields**.
> - The new functionality is not enabled by default, and can be enabled with a checkbox in the preferences screen. This also enables
>   the new apkg import/export code introduced in 2.1.52. If you use the Special Fields add-on, you will need to disable this option
>   in order to use the add-on.

:open_mouth: :tada: 

Possible removal of CrowdAnki as a dependency...? :eyes:

**Comment by aplaice** [comment]
Quickly testing the functionality, it seems that this modified CSV import/export format still doesn't allow importing/exporting the templates of the note model (so if a user doesn't have the Ultimate Geography note type, they won't be able to get it, and we also wouldn't be able to update the templates). :/
